plugins/repository-s3/src/test/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStoreContainerTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.aws.blobstore;
+
+import com.amazonaws.Protocol;
+import com.amazonaws.services.s3.AmazonS3;
+import org.elasticsearch.cloud.aws.InternalAwsS3Service;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.repositories.ESBlobStoreContainerTestCase;
+
+import java.io.IOException;
+import java.util.Locale;
+
+public class S3BlobStoreContainerTests extends ESBlobStoreContainerTestCase {
+    protected BlobStore newBlobStore() throws IOException {
+        int maxRetries = 5;
+        String region = null;
+        Settings settings = Settings.EMPTY;
+
+        InternalAwsS3Service service = new InternalAwsS3Service(settings);
+        AmazonS3 client = service.client(null, Protocol.HTTPS, region, null, null, maxRetries);
+        String bucket = randomAsciiOfLength(randomIntBetween(1, 10)).toLowerCase(Locale.ROOT);
+
+        return new S3BlobStore(settings, client, bucket, region, false,
+            new ByteSizeValue(100), maxRetries, "public-read-write", "standard");
+    }
```

Comment:
True, which means you would need a mock implementation of
DefaultS3OutputStream as well - it just needs to be a mock of
java.io.OutputStream that takes your mock AmazonS3 client as a private
member and the write operations of the OutputStream just take the
bytes/input stream and store them in the map of blob data in your mock
AmazonS3.

What do you think?

On Mon, Jun 20, 2016 at 12:10 AM, gfyoung <notifications@github.com> wrote:

> In
> plugins/repository-s3/src/test/java/org/elasticsearch/cloud/aws/blobstore/S3BlobStoreContainerTests.java
> <https://github.com/elastic/elasticsearch/pull/18815#discussion_r67634349>
> :
>
> > +import java.io.IOException;
> > +import java.util.Locale;
> > +
> > +public class S3BlobStoreContainerTests extends ESBlobStoreContainerTestCase {
> > +    protected BlobStore newBlobStore() throws IOException {
> > +        int maxRetries = 5;
> > +        String region = null;
> > +        Settings settings = Settings.EMPTY;
> > +
> > +        InternalAwsS3Service service = new InternalAwsS3Service(settings);
> > +        AmazonS3 client = service.client(null, Protocol.HTTPS, region, null, null, maxRetries);
> > +        String bucket = randomAsciiOfLength(randomIntBetween(1, 10)).toLowerCase(Locale.ROOT);
> > +
> > +        return new S3BlobStore(settings, client, bucket, region, false,
> > +            new ByteSizeValue(100), maxRetries, "public-read-write", "standard");
> > +    }
>
> Yes, I know, but go a little bit deeper, and you'll see that S3BlobStore
> does nothing besides help to instantiate the stream.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/18815/files/5ecfabc0dffffcbec1cbf7c5c7db0e3ab05a8c07#r67634349>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABjkQQiBXpZ9pD94hIAC0G_7Pkyhvw1-ks5qNhKbgaJpZM4Iy-tw>
> .
>